### ISSUE TITLE
[BUG] MAUI refusing to build for iOS with package

### ISSUE THREAD

**Issue by NickD-Callaway** [Azure/azure-sdk-for-net#30156]
### Library name and version

Azure.Identity 1.6.0

### Describe the bug

I am currently having an issue with building a MAUI project with the Azure.Identity 1.6.0 library installed. When I try to build it, I get a 'clang++ exited with code 1' error in the console and refuses to go any further. It works building for Windows.

### Expected behavior

It should be able to build for the iOS platform

### Actual behavior

I refuses to build for the iOS platform with the error 'clang++ exited with code 1'

### Reproduction Steps

1. Create a new default MAUI project
2. Import the Azure.Identity 1.6.0 library
3. Make sure you are paired to a Mac device
4. Switch the Build Device to an iOS simulator or Remote Device
5. Run the project on simulator or remote device
6. The error will happen in the build process

Attached is a project I quickly got together: [MauiAzureTest.zip](https://github.com/Azure/azure-sdk-for-net/files/9200939/MauiAzureTest.zip)

### Environment

Using:
Visual Studio 2022 Preview 3.0 (Version 17.3.0 Preview 3.0)
MAUI version 6.0.408
.NET 6

**Comment by jsquire** [comment]
Hi @NickD-Callaway.   Thank you for reaching out and we regret that you're experiencing difficulties.   I was able to reproduce the behavior that you're seeing with the project that you attached using `dotnet build` directly on MacOS.   However, I trigger the same error with and without the package reference to `Azure.Identity`.

When I create a new default Maui project, add a reference to `Azure.Identity` and build, I do not receive errors.   I'm able to run the application natively on the Mac, on an iOS simulator, and an Android simulator.   
That said, I cannot seem to isolate the issue to something specific in your project.  I've run through a diff on every file and don't see anything outside of GUIDs and some formatting differences.   I even copied my project file over, removed all the `.vs` and `*.user` customizations, and still run into the issue.

I'm reasonably sure that your build error is unrelated to the Azure package reference.   I believe it has something to do with the message `ld: framework not found System`, but I can't reasonably speculate about a root cause.  **I believe your best path forward would be to report this to the Maui team, as they will have the expertise to investigate further and advise.**

In case it helps, I'm including the full output for `dotnet build` when run against your project:

```text
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Detected signing identity:
          
    Bundle Id: com.companyname.mauiazuretest
    App Id: com.companyname.mauiazuretest
  Detected signing identity:
          
    Bundle Id: com.companyname.mauiazuretest
    App Id: com.companyname.mauiazuretest
  MauiAzureTest -> /Users/jesse/src/scratch/MauiAzureTest/bin/Debug/net6.0-maccatalyst/maccatalyst-x64/MauiAzureTest.dll
  MauiAzureTest -> /Users/jesse/src/scratch/MauiAzureTest/bin/Debug/net6.0-ios/iossimulator-x64/MauiAzureTest.dll
  MauiAzureTest -> /Users/jesse/src/scratch/MauiAzureTest/bin/Debug/net6.0-android/MauiAzureTest.dll
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Tool xcrun execution finished (exit code = 1).
          
  ld: framework not found System
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.328/targets/Xamarin.Shared.Sdk.targets(1219,3): error : clang++ exited with code 1 [/Users/jesse/src/scratch/MauiAzureTest/MauiAzureTest.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.328/targets/Xamarin.Shared.Sdk.targets(1219,3): error :          [/Users/jesse/src/scratch/MauiAzureTest/MauiAzureTest.csproj]

Build FAILED.
```

**Comment by NickD-Callaway** [comment]
Thanks for you response @jsquire ! My next course of action is to get ahold of the MAUI team.

I spent some time just now trying to use my Mac to do the same thing with no success. Same issue. I created the base MAUI project using VS for Mac 17.3 Preview, imported the Azure.Identity library, and it still gives me the clang++ error. I will bring this up with them and see what they have to say about the issue.

Thanks!!
